### PR TITLE
Fixing code smells

### PR DESCRIPTION
**Pull request by nhurtado**
Fixing code smells start from the "Removed the use of 'global' inside functions" commit.
Fixed 4 code smells of global data, by using a singleton class.

### CODE REVIEW

**Comment by nhurtado** [comment]
[Check and correct the list of code smells](https://trello.com/c/4Q2uU3hA/23-check-and-correct-the-list-of-code-smells)

**Comment by nhurtado** [comment]
Fixed 4 different Loops. Avoided using Fors by using Filter and Map.

**Comment by cefunck** [comment]
Code smells correction ready.

File: BlockModel.py [hunk 1]
```diff
@@ -75,25 +75,28 @@ def total_weight(self):
         total_weight = sum(block.weight for block in self.blocks)
         return total_weight
 
+    def __grade_weight(self,grade, grade_type):
+        weight = 0
+        for block in self.blocks:
```
Comment (vreyn) [inline BlockModel.py]: You can improve this further by removing this smelly loops

**Pull request merged**